src/main/java/core/basesyntax/service/operation/BalanceOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitsForSale;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void updateDataBase(FruitTransaction transaction) {
+        FruitsForSale fruitsForSale = new FruitsForSale();
+        if (fruitsForSale.isNotForSale(transaction.getFruit())) {
+            throw new RuntimeException("We don't sell such fruit: " + transaction.getFruit());
```

Comment:
The error message could be more informative by specifying the transaction details or context. This would help in debugging and understanding which transaction caused the error.